File: ABC/031/C.rb
N = gets.to_i
a = gets.split.map(&:to_i)

ans = -10 ** 5
(0..N-1).each do |i|
  aoki = []
  (0..N-1).each do |j|
    if i == j
      next
    end
    left = [i,j].min
    right = [i,j].max
    t = a.[](left..right)
    even = t.select.with_index { |_, i| i.odd? }
    aoki.push(even.inject(:+) )
  end
  j = aoki.index(aoki.max)
  left = [i,j].min
  right = [i,j].max
  t = a.[](left..right)
  odd = t.select.with_index { |_, i| i.even? }
  ans = [ans,odd.inject(:+)].max
end

puts ans

File: ABC/031/C.py
def main():
    N = int(input())
    A = list(map(int, input().split()))
    ans = -(10 ** 5)
    for i in range(N):
        aoki = []
        for j in range(N):
            if i == j: continue

            left, right = min(i,j), max(i,j)
            T = A[left: right+1]
            aoki.append(sum(T[1::2]))

        j = aoki.index(max(aoki))
        left, right = min(i, j), max(i, j)
        T = A[left: right+1]
        ans = max(ans, sum(T[0::2]))
                        
    print(ans)
                        

if __name__ == "__main__":
    main()